src/app/qgslayertreeviewfilterindicator.cpp

```diff
@@ -58,6 +59,10 @@ QString QgsLayerTreeViewFilterIndicatorProvider::tooltipText( QgsMapLayer *layer
   if ( rlayer && rlayer->dataProvider() && rlayer->dataProvider()->supportsSubsetString() )
     filter = rlayer->subsetString();
 
+  QgsPointCloudLayer *pclayer = qobject_cast<QgsPointCloudLayer *>( layer );
```

Comment:
I think it's time to rework this logic to use a switch on layer->type()

Reply:
We won't get rid of the casts though. We could instead add virtual subsetString() - subsetStringChanged signal to QgsMapLayer instead of the the subclasses and stop caring about layer type?

Reply:
I'm fine with the casts, it's just the repeated "if (dynamic_cast ) ... " blocks which aren't ideal. A switch on layer type will clean this up and avoid the multiple casts (and also give us warnings if/when more layer types are added in future)